authors/apps/articles/tests/test_rate_article.py

```diff
@@ -0,0 +1,170 @@
+            'user': {
+                'username': 'user2',
+                'email': 'evajohnson715s@gmail.com',
+                'password': 'somepass12345',
+            }
+        }
+        resp = self.client.post(
+            self.signup_url,
+            self.user_details,
+            format='json')
+        res = self.client.post(
+            self.signup_url,
+            self.user_details_2,
+            format='json')
+        self.token_2 = res.data['token']
+        self.token = resp.data['token']
+        self.email = "test_user@gmail.com"
+        self.name = "test"
+        self.user = User(username=self.name, email=self.email)
+        self.user.set_password("@Winners11")
+        self.user.save()
+        self.user_id = User.objects.get(email=self.email).pk
+        self.slug = "this-is-a-question"
+        title = "this is a question"
+        description = "this is a description"
+        body = "this is a body"
+        author = self.user
+        article = Article(
+            author=author,
+            slug=self.slug,
+            body=body,
+            title=title,
+            description=description)
+        article.save()
+        self.rate_details = {
+            "user": {
+                "slug": self.slug,
+                "rate": 3
+            }
+        }
+        self.rate_url = "http://localhost:8000/api/article/"\
+            + self.slug + "/rate/"
+        self.view_rates_url = "http://localhost:8000/api/article/rate/"
+
+    def test_rate_article_without_token(self):
+        """
+        test weather rate article without token will fail.
+        """
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'Authentication credentials were not provided.', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_rate_article(self):
+        """
+        test rate article.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'sucessfully rated', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_rate_article_not_found(self):
+        """
+        test weather rate article without article will fail.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        self.rate_details["user"]["slug"] = "-ss-dd-dd-ff"
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'Article not found', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
I think it should be a `204` for no content since we are testing for Article not found
cc: @wrotich @abadojack @evamaina @TheSteelGuy @Eyansky